app/models/user.rb

```diff
@@ -208,44 +208,57 @@ def self.upload_user_list(user_class, user_list, encoding)
     imported = nil
     parsed[:invalid_records] = ''
     User.transaction do
-      imported = user_class.import user_columns, users, on_duplicate_key_update: {
-        conflict_target: [:user_name],
-        columns: [:last_name, :first_name, :section_id, :email, :id_number, :display_name, :time_zone]
-      }
-      User.where(id: imported.ids).each do |user|
-        if user_class == Ta
-          # This will only trigger before_create callback in ta model, not after_create callback
-          user.run_callbacks(:create) { false }
+      users.each do |user|
```

Comment:
Hi David, thanks so much for all the help! In my newest commit, I tried to implement this, however, it would cause imported to always be nil. Unsure why this is happening, but I will look into it later.